asciidoctorj-core/src/main/java/org/asciidoctor/extension/Processor.java

```diff
@@ -16,25 +17,29 @@
 
 public class Processor {
 
-	protected RubyHash config;
-    protected Ruby rubyRuntime;
+	protected Map<String, Object> config;
 
     public Processor(Map<String, Object> config) {
-        this.rubyRuntime = JRubyRuntimeContext.get();
-        this.config = RubyHashUtil.convertMapToRubyHashWithSymbols(rubyRuntime, config);
+        this.config = config;
     }
 
-    public void update_config(Map<Object, Object> config) {
+    public void update_config(Map<String, Object> config) {
     	this.config.putAll(config);
     }
     
-    public Map<Object, Object> getConfig() {
+    public Map<String, Object> getConfig() {
     	return this.config;
     }
 
-    public RubyHash setConfig(Map<String, Object> config) {
-        this.config = RubyHashUtil.convertMapToRubyHashWithSymbols(rubyRuntime, config);
```

Comment:
This was still a gap.
If someone sets the config after the ProcessorProxy has connected to this Proxy, the config must not be simply replaced because this would bring the config maps out of sync between Java and Ruby.
I wonder if it makes sense at all to change the config after an extension was is registered.
In particular as for example the InlineMacroProcessor does further actions depending on the config in the initializer.